src/content/1.7/development/components/grid-component.md

```diff
@@ -7,229 +7,404 @@ weight: 2
+$searchCriteria->getLimit();    // 10
+```
+
+When creating Search Criteria you can skip some or all it's data. You can leave default `null` values for order way
+and order by to disable sorting or offset and limit to disable pagination.
 
 ```php
-interface DoctrineQueryBuilderInterface
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteria;
+
+// both sorting, pagination and filtering are disabled with this search criteria
+$emptySearchCriteria = new SearchCriteria();
+
+// only pagination is set
+// that means sorting (and filters as it's an empty array) will be disabled for search criteria
+$emptySortingSearchCriteria = new SearchCriteria(
+    [],
+    null,
+    null,
+    2,
+    10
+);
+```
+
+## Grid Data
+
+The final part of Grid component is data. Grid Data is stored in `PrestaShop\PrestaShop\Core\Grid\Data\GridData`. 
+
+### Creating Grid Data
+
+Grid component does not create Grid Data directly but instead relies on `PrestaShop\PrestaShop\Core\Grid\Data\Factory\GridDataFactoryInterface`.
+PrestaShop provides you with `DoctrineGridDataFactory` implementation out of the box which supports retrieving data from
+MySQL database using Doctrine. However, if you need to load data from REST API, Elasticsearch or any other data store,
+you can implement your own Grid Data factory.
+
+We will be using `DoctrineGridDataFactory` to create data for our Grid. When using `DoctrineGridDataFactory` you have
+to implement `DoctrineQueryBuilderInterface` which will be used by data factory to build Doctrine queries.
+
+{{% notice note %}}
+When implementing `DoctrineQueryBuilderInterface` it is recommended to use `PrestaShop\PrestaShop\Core\Grid\Query\AbstractDoctrineQueryBuilder`
+as it provides access to Doctrine `Connection` and database tables prefix.
+{{% /notice %}}
+
+```php
+use PrestaShop\PrestaShop\Core\Grid\Query\AbstractDoctrineQueryBuilder;
+
+final class ProductQueryBuilder extends AbstractDoctrineQueryBuilder
 {
     /**
-     * Get query that searches grid rows
-     *
-     * @param SearchCriteriaInterface|null $searchCriteria
-     *
-     * @return QueryBuilder
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @var int
      */
-    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria = null);
+    private $contextShopId;
+
     /**
-     * Get query that counts grid rows
-     *
-     * @param SearchCriteriaInterface|null $searchCriteria
-     *
-     * @return QueryBuilder
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param int $contextLangId
+     * @param int $contextShopId
      */
-    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria = null);
+    public function __construct(Connection $connection, $dbPrefix, $contextLangId, $contextShopId)
+    {
+        parent::__construct($connection, $dbPrefix);
 
-```
+        $this->contextLangId = $contextLangId;
+        $this->contextShopId = $contextShopId;
+    }
```

Comment:
I love that you presented an example with language and shop context!!
which makes me think, maybe we could add an abstract class which does that?

Reply:
I love this idea, this will be help many people :)